crates/bevy_ecs/macros/src/component.rs

```diff
@@ -121,3 +150,7 @@ fn storage_path(bevy_ecs_path: &Path, ty: StorageTy) -> TokenStream2 {
 
     quote! { #bevy_ecs_path::component::StorageType::#storage_type }
 }
+
+fn hook_function(hook: TokenStream2, function: Option<Meta>) -> Option<TokenStream2> {
```

Comment:
Bikeshed: I think `register_hook` might be a bit clearer

Reply:
Changed to `hook_register_function_call`, because I think it explains better what this function returns